powersimdata/design/investment/investment_costs.py

```diff
@@ -40,21 +40,29 @@ def append_keep_index_name(df1, other, *args, **kwargs):
     return new_df
 
 
-def calculate_ac_inv_costs(scenario, sum_results=True, exclude_branches=None):
+def calculate_ac_inv_costs(
+    scenario,
+    sum_results=True,
+    exclude_branches=None,
+    base_grid=None,
+):
     """Calculate cost of upgrading AC lines and/or transformers in a scenario.
     NEEM regions are used to find regional multipliers.
 
     :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
     :param bool sum_results: whether to sum data frame for each branch type. Defaults to
         True.
+    :param powersimdata.input.grid.Grid base_grid: a Grid to compare against. If None,
+        the grid model and interconnect from the ``scenario`` are used to instantiate a
+        corresponding unmodified Grid.
     :return: (*dict*) -- keys are {'line_cost', 'transformer_cost'}, values are either
         float if ``sum_results``, or pandas Series indexed by branch ID.
         Whether summed or not, values are $USD, inflation-adjusted to today.
     """
-
-    base_grid = Grid(
-        scenario.info["interconnect"].split("_"), source=scenario.info["grid_model"]
-    )
+    if base_grid is None:
+        base_grid = Grid(
+            scenario.info["interconnect"].split("_"), source=scenario.info["grid_model"]
+        )
```

Comment:
if `base_grid` is not None, do we need to make sure that the grid model is the same than the one in the `scenario` object?

Reply:
I think the purpose of having user defined `base_grid` here is to provide the capability of comparing the difference between the grid in the `scenario` and any grid instead of the `base_grid` in PowerSimdata only.

Reply:
@BainanXia, my question is can we have a usa_tamu model in the `scenario` object, i.e.,`scenario.get_grid().grid_model == "usa_tamu"`), and a "hifld" model for the grid that is passed, i.e., `base_grid.grid_model == "hifld"`

Reply:
I agree, at the very least we want a check that the grid models are the same. We _may_ also want a check that the interconnections match too, or we need to be sure that we have well-defined behavior for comparing just the overlapping parts of two Grids (e.g. comparing a Western expanded scenario to a USA base scenario).

Reply:
Done. We now check for the matching grid model, and we should be able to handle interconnections via the reindexing logic that we add in #507.